src/BloomTests/PretendRequestInfo.cs

```diff
@@ -14,11 +14,13 @@ public class PretendRequestInfo : IRequestInfo
 		public long StatusCode;
 		public string StatusDescription;
 
-		public PretendRequestInfo(string url)
+		public PretendRequestInfo(string url, bool forPrinting)
```

Comment:
You could avoid a lot of changes and some extra code by declaring this as "bool forPrinting = false"